src/Symfony/Component/Translation/Tests/Util/ArrayConverterTest.php

```diff
@@ -69,6 +69,20 @@ public static function messagesData()
                     ],
                 ],
             ],
+            [
+                // input
+                [
+                    'foo.' => 'foo.',
+                    'bar.' => 'bar.',
+                    'foo bar.' => 'foo bar.',
+                ],
+                // expected output
+                [
+                    'foo' => 'foo.',
```

Comment:
As mentioned [here](https://github.com/symfony/symfony/issues/51536#issuecomment-1703371496), the periods should not be trimmed, as it modifies the keys and leads to inconsistent output files with the extracted translations. 

We could also test that periods at the beginning of the key don't create empty keys in the tree :smiley:

Reply:
Nice catch.

In order to cover both scenarios a new `getKeyParts()` function was added, this ensures the message keys contain the expected strings, and, additional test cases were added with the suggestions done.

Thanks in advance.